sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -432,7 +442,13 @@ static void _send(ng_pktsnip_t *pkt, bool prep_hdr)
     else {
         ng_ipv6_addr_t *next_hop = NULL;
 
-        next_hop = &hdr->dst;   /* TODO: next hop determination */
+#ifdef MODULE_NG_IPV6_EXT_RH
+        next_hop = ng_ipv6_ext_rh_next_hop(hdr);
```

Comment:
You could e.g. save a pointer to the routing header in the incoming ipv6 packet and change the parameter of `ng_ipv6_ext_rh_next_hop` to be of type `ng_ipv6_ext_rh_t`. This way you would not need to search for this specific header.